tree/dataframe/src/RSqliteDS.cxx

```diff
@@ -60,11 +388,22 @@ constexpr char const *RSqliteDS::fgTypeNames[];
 RSqliteDS::RSqliteDS(std::string_view fileName, std::string_view query)
    : fDb(nullptr), fQuery(nullptr), fNSlots(0), fNRow(0)
 {
+   static bool isRegistered = Register();
    int retval;
 
-   retval = sqlite3_open_v2(std::string(fileName).c_str(), &fDb, SQLITE_OPEN_READONLY | SQLITE_OPEN_NOMUTEX, nullptr);
-   if (retval != SQLITE_OK)
-      SqliteError(retval);
+   // Opening the layer
+   if (IsURL(fileName)) {
+      if (!isRegistered)
+         throw std::runtime_error("Error");
+      retval = sqlite3_open_v2(std::string(fileName).c_str(), &fDb, SQLITE_OPEN_READONLY | SQLITE_OPEN_NOMUTEX, gSQliteVfsName);
```

Comment:
Please don't use `string_view` if the contents will be copied anyway.

Reply:
I changed it but I wonder if you are sure about this. It makes the interface slightly inconsistent (one parameter is a `string`, the other one a `string_view`) and the interface becomes dependent on the implementation details.

Reply:
there are diverging opinions on this in the team. Philippe (and maybe Axel) is very much in favor of always exposing string_view in user-facing APIs, I'm somewhat undecided, Guilherme is very much against it especially when it results in strings being constructed in the body of the function.

in this case I don't think it really matter as it's an internal function (datasources are always constructed by factory functions, which incidentally imo should take `string_view`s)

Reply:
> when it results in strings being constructed in the body of the function.

Indeed if/when the input is copied *as is* then it make senses, otherwise having the argument being an std::string makes a useless/unnecessary copy.

Reply:
In this context, the 'efficient' solution is to have 3 overload (const char*, const std::string& and std::string_view) and to forward the later two to the first one [The blocker here is that you end up calling a C interface and thus you need a C-style string but you do *not* need a copy of the string.]

Reply:
If/since it is an internal routine it might make sense to 'just' use "const char*" and do the necessary conversion upstream.

Reply:
Let me just explain my view. I am against `string_view` because it creates problems when it's used in user interfaces and can be either our backported implementation or the standard one, depending on configuration flags. This is a point where we can break ABI (or even API if our backport deviates from the standard, which it did for a while). It's a similar problem to using SIMD types in the interfaces, which I also strongly oppose. Using `string_view` would not be a problem if our baseline was C++17, but since it's C++11, it is. In most places where `string_view` has been used, it brings basically no benefits, because its contents are copied anyway. It's in those places I'm advocating for not using it. I think that we've had many more cons than pros since the introduction of this backport into ROOT: https://root-forum.cern.ch/search?q=string_view. That said, ultimately the choice is of the implementer, so take my comment above (the first one) only as a suggestion.

Reply:
You are discussing a design pattern that is beyond this PR. To discuss it, please send an email to rootdev@cern.ch instead.

ROOT must be built and used (as includes and libraries) with a consistent configuration; the C++ standard is part of this and people get it from `root-config --cflags`. I understand that you don't like that, but let's discuss that at rootdev. Thanks!